src/Transport/Topology/Topologies/ForwardingTopologySectionManager.cs

```diff
@@ -102,28 +112,28 @@ public TopologySection DeterminePublishDestination(Type eventType)
                     BuildTopicBundles(namespaces, addressingLogic);
                 }
 
-                return new TopologySection()
+                return new TopologySection
                 {
-                    Entities = SelectSingleRandomTopicFromBundle(topics),
+                    Entities = SelectFirstTopicFromBundle(topics),
                     Namespaces = namespaces
                 };
             });
         }
 
-        IEnumerable<EntityInfo> SelectSingleRandomTopicFromBundle(List<EntityInfo> entityInfos)
+        IEnumerable<EntityInfo> SelectFirstTopicFromBundle(List<EntityInfo> entityInfos)
         {
-            var index = randomGenerator.Next(0, entityInfos.Count);
+            const int index = 1;
             var selected = entityInfos[index];
 
-            return entityInfos.Where(i => i.Path == selected.Path);
+            yield return entityInfos.First(i => i.Path == selected.Path);
```

Comment:
Isn't this overly complex? We already have the first entry selected by using the list indexer. Can't we just return that? Why is the additional First with the condition for the selection needed?